src/constant.h

```diff
@@ -7,594 +7,598 @@
 // you change anything below, without understanding EXACTLY how the game
-// uses the number, the program may stop working correctly.  Modify the
+// uses the number, the program may stop working correctly. Modify the
 // constants at your own risk. -JEW-
 
 // message line location
-#define MSG_LINE 0
+constexpr int32_t MSG_LINE = 0;
 
 // number of messages to save in a buffer
-#define MAX_SAVE_MSG    22   // How many messages to save -CJS-
+constexpr int32_t MAX_SAVE_MSG = 22; // How many messages to save -CJS-
 
 // Dungeon size parameters
-#define MAX_HEIGHT      66   // Multiple of 11; >= 22
-#define MAX_WIDTH      198   // Multiple of 33; >= 66
-#define SCREEN_HEIGHT   22
-#define SCREEN_WIDTH    66
-#define QUART_HEIGHT    (SCREEN_HEIGHT / 4)
-#define QUART_WIDTH     (SCREEN_WIDTH / 4)
+constexpr int32_t MAX_HEIGHT = 66; // Multiple of = 11; >= = 22
+constexpr int32_t MAX_WIDTH = 198; // Multiple of = 33; >= = 66
+constexpr int32_t SCREEN_HEIGHT = 22;
+constexpr int32_t SCREEN_WIDTH = 66;
+constexpr int32_t QUART_HEIGHT = (SCREEN_HEIGHT / 4);
+constexpr int32_t QUART_WIDTH = (SCREEN_WIDTH / 4);
 
 // Dungeon generation values
 // Note: The entire design of dungeon can be changed by only slight adjustments here.
-#define DUN_TUN_RND               9   // 1/Chance of Random direction
-#define DUN_TUN_CHG              70   // Chance of changing direction (99 max)
-#define DUN_TUN_CON              15   // Chance of extra tunneling
-#define DUN_ROO_MEA              32   // Mean of # of rooms, standard dev2
-#define DUN_TUN_PEN              25   // % chance of room doors
-#define DUN_TUN_JCT              15   // % chance of doors at tunnel junctions
-#define DUN_STR_DEN               5   // Density of streamers
-#define DUN_STR_RNG               2   // Width of streamers
-#define DUN_STR_MAG               3   // Number of magma streamers
-#define DUN_STR_MC               90   // 1/x chance of treasure per magma
-#define DUN_STR_QUA               2   // Number of quartz streamers
-#define DUN_STR_QC               40   // 1/x chance of treasure per quartz
-#define DUN_UNUSUAL             300   // Level/x chance of unusual room
+constexpr int32_t DUN_TUN_RND = 9; // = 1/Chance of Random direction
+constexpr int32_t DUN_TUN_CHG = 70; // Chance of changing direction (99 max)
+constexpr int32_t DUN_TUN_CON = 15; // Chance of extra tunneling
+constexpr int32_t DUN_ROO_MEA = 32; // Mean of # of rooms, standard dev2
+constexpr int32_t DUN_TUN_PEN = 25; // % chance of room doors
+constexpr int32_t DUN_TUN_JCT = 15; // % chance of doors at tunnel junctions
+constexpr int32_t DUN_STR_DEN = 5; // Density of streamers
+constexpr int32_t DUN_STR_RNG = 2; // Width of streamers
+constexpr int32_t DUN_STR_MAG = 3; // Number of magma streamers
+constexpr int32_t DUN_STR_MC = 90; // = 1/x chance of treasure per magma
+constexpr int32_t DUN_STR_QUA = 2; // Number of quartz streamers
+constexpr int32_t DUN_STR_QC = 40; // = 1/x chance of treasure per quartz
```

Comment:
Three search/replace errors. Should be:
`// 1/Chance of Random direction`
and
`// 1/x chance of treasure per magma`
and
`// 1/x chance of treasure per quartz`